core/services/workflows/workflow_spec_factory.go

```diff
@@ -0,0 +1,51 @@
+package workflows
+
+import (
+	"crypto/sha256"
+	"errors"
+	"fmt"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/sdk"
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/job"
+)
+
+var ErrInvalidWorkflowType = errors.New("invalid workflow type")
+
+type SDKWorkflowSpecFactory interface {
+	GetSpec(rawSpec, config []byte) (sdk.WorkflowSpec, error)
+	GetRawSpec(wf string) ([]byte, error)
```

Comment:
Two functions will allow us to potentially have WASM files loaded in different ways.  I figured that EOD it will store something on the box, so there would be a file version.  There may also be one that resolves HTTP at runtime, I figured that could download embed the WASM file version, but override GetRawSpec to do the HTTP download first.

Reply:
Also, it could allow caching later if we watned.